light_node/src/main.rs

```diff
@@ -82,6 +84,8 @@ fn create_logger(env: &crate::configuration::Environment) -> Logger {
 fn block_on_actors(env: &crate::configuration::Environment, identity: Identity, actor_system: ActorSystem, init_info: TezosStorageInitInfo, persistent_storage: PersistentStorage, protocol_commands: IpcCmdServer, protocol_events: IpcEvtServer, protocol_runner_run: Arc<AtomicBool>, log: Logger) {
     let mut tokio_runtime = Runtime::new().expect("Failed to create tokio runtime");
 
+    let ctx_storage = Arc::new(RwLock::new(SkipList::new(0, &persistent_storage).expect("failed to initialize context storage")));
```

Comment:
We should not pass skip_list storage as a separate variable, but we should extent `PersistentStorage` to include it.

Reply:
This is non-trivially solvable cyclical dependency problem, as it requires refactorization of whole `storage` folder.
`SkipList` requires many traits in `storage` crate, but extending `PersistentStorage` to include `SkipList` would make cyclical dependency between `storage` and `skip_list` crate.

Reply:
Skip list does not require the whole `PersistentStorage`. You can pass only `Arc<rocksdb::DB>` to it, that will solve dependency problem.

Reply:
I see you did the `Arc<rocksdb::DB>` in https://github.com/simplestaking/tezedge/commit/c97f893d88a5f82935933f4ec04a5ba42f4436ad
Now the cyclical dependency should be there there.

Reply:
Cyclical dependency happens, because `PersistentStorage` struct is in same crate as many traits like `BincodeEncoded` and `KeyValueStoreWithSchema`, which are essential to database backed structure, so even if `SkipList` does not directly requries the `PersistentStorage` the cyclical dependency still holds. Only way to somehow fix this, is to move `PersistentStorage` somewhere else.

Reply:
We could move `SkipList` into a separate module inside of the `storage` package.

Reply:
fixed in c694da1